src/event-bus.ts

```diff
@@ -36,10 +36,6 @@ export class EventBus extends ObservableBus<IEvent> implements IEventBus {
     this._publisher.publish(event);
   }
 
-  ofType<T extends IEvent>(event: T & { name: string }) {
-    return this.ofEventName(event.name);
-  }
-
```

Comment:
I believe that this method should be actually removed and replaced with pipeable operator, `.pipe(ofType(...))`, same as here https://ngrx.io/guide/effects

Reply:
Agreed! However, the type of `events$` in Saga is `EventObservable<IEvent>`, which does not expose `pipe` (the class implementing it, `ObservableBus`, does; and so does `EventBus` (as it extends `ObservableBus`)).

Yet I could indeed replace the operator `filter` by `ofType`  in in `ObservableBus`:

```typescript
  ofType(...types: Type<T>[]): Observable<T> {
    return this.pipe(
      filter(event => !isEmpty(types.filter(type => event instanceof type))),
    );
  }
```

What do you think?

Use case (with the saga):

```typescript
fakeMethodBuildingOneSaga() {
  return (events$: EventObservable<IEvent>): Observable<ICommand> =>
    // the only exposed method for `events$` is `ofType`
    events$.ofType(...[EventType1, EventType2, EventType3]).pipe(
      // operators
    );
}
```

Reply:
I believe that the whole `EventObservable` class could be removed in return for a normal pipeable function used together with a pure `Observable`

Reply:
Changed/rebased a bit :grin:

Proposal (with the existing code):

```typescript
 buildCatsSaga() {
    return (events$: Observable<IEvent>): Observable<ICommand> =>
      events$.pipe(
        instanceOf(...this.targetEventTypes),
        // ...
        map((value: something) => new CatCommand(workerId)),
      );
  }
```

Suggestion most welcome. I took the freedom to add Jest to the package :see_no_evil: